texar/modules/classifiers/xlnet_classifier.py

```diff
@@ -0,0 +1,173 @@
+            encoder_hparams = utils.dict_fetch(
+                hparams, XLNetEncoder.default_hparams())
+            if encoder_hparams is not None:
+                encoder_hparams['name'] = None
+            self._encoder = XLNetEncoder(
+                pretrained_model_name=pretrained_model_name,
+                cache_dir=cache_dir,
+                hparams=encoder_hparams)
+            if self._hparams.use_projection:
+                self.projection = layers.get_layer(hparams={
+                    "type": "Dense",
+                    "kwargs": {
+                        "units": self._encoder.output_size
+                    }
+                })
+
+            # summary type
+            if self._hparams.summary_type == 'last':
+                self.summary_op = lambda output: output[:, -1]
+            elif self._hparams.summary_type == 'first':
+                self.summary_op = lambda output: output[:, 0]
+            elif self._hparams.summary_type == 'mean':
+                self.summary_op = lambda output: tf.reduce_mean(output, axis=1)
+            elif self._hparams.summary_type == 'attn':
+                raise NotImplementedError
+            else:
+                raise ValueError(
+                    f"Unsupported summary type {self._hparams.summary_type}")
+
+            # Creates an dropout layer
+            drop_kwargs = {"rate": self._hparams.dropout}
+            layer_hparams = {"type": "Dropout", "kwargs": drop_kwargs}
+            self._dropout_layer = layers.get_layer(hparams=layer_hparams)
+
+            # Creates an additional classification layer if needed
+            self._num_classes = self._hparams.num_classes
+            if self._num_classes <= 0:
+                self._logit_layer = None
+            else:
+                logit_kwargs = self._hparams.logit_layer_kwargs
+                if logit_kwargs is None:
+                    logit_kwargs = {}
+                elif not isinstance(logit_kwargs, HParams):
+                    raise ValueError(
+                        "hparams['logit_layer_kwargs'] must be a dict.")
+                else:
+                    logit_kwargs = logit_kwargs.todict()
+                logit_kwargs.update({"units": self._num_classes})
+                if 'name' not in logit_kwargs:
+                    logit_kwargs['name'] = "logit_layer"
+
+                layer_hparams = {"type": "Dense", "kwargs": logit_kwargs}
+                self._logit_layer = layers.get_layer(hparams=layer_hparams)
+
+    @staticmethod
+    def default_hparams():
+        hparams = XLNetEncoder.default_hparams()
+        hparams.update({
+            "num_classes": 2,
+            "logit_layer_kwargs": None,
+            "summary_type": "last",
+            "dropout": 0.1,
+            "use_projection": True,
+            "name": "xlnet_classifier"
+        })
+        return hparams
+
+    def _build(self, token_ids, segment_ids=None, input_mask=None, mode=None):
```

Comment:
add docstring.  `_build` will be shown in the doc (e.g., https://texar.readthedocs.io/en/latest/code/modules.html#texar.modules.Conv1DClassifier._build)

Reply:
Done! [3448a9c]